generators/server/templates/src/main/java/package/config/GatewaySwaggerResourcesProvider.java.ejs

```diff
@@ -0,0 +1,115 @@
+package <%= packageName %>.config;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.util.function.Tuple2;
+import springfox.documentation.oas.annotations.EnableOpenApi;
+import springfox.documentation.swagger.web.SwaggerResource;
+import springfox.documentation.swagger.web.SwaggerResourcesProvider;
+import tech.jhipster.config.JHipsterConstants;
+
+/**
+ * Retrieves all registered microservices Swagger resources.
+ */
+@Component
+@Primary
+@Profile(JHipsterConstants.SPRING_PROFILE_API_DOCS)
+@Configuration
+@EnableOpenApi
+public class GatewaySwaggerResourcesProvider implements SwaggerResourcesProvider {
+
+    private final RouteLocator routeLocator;
+
+    @Qualifier("swaggerResources")
+    private final SwaggerResourcesProvider swaggerResources;
```

Comment:
Maybe you could name this `gatewaySwaggerResources` so it will be easier to understand that it's the swagger resources of the gateway itself